.evergreen/build_snapshot_rpm.sh

```diff
@@ -0,0 +1,173 @@
+  rm -f  ${spec_file}
+fi
+/usr/bin/curl -L --retry 5 -o "${spec_file}" "${spec_url}"
+if [ "${?}" != "0" -o ! -f "${spec_file}" ]; then
+  echo "Could not retrieve spec file from URL: ${spec_url}"
+  exit 1
+fi
+if [ -f .evergreen/spec.patch ]; then
+  patch -d .. -p0 -i $(pwd)/.evergreen/spec.patch
+fi
+
+changelog_package=$(rpmspec --srpm -q --qf "%{name}" ${spec_file})
+if [ "${package}" != "${changelog_package}" ]; then
+  echo "This script is configured to create snapshots for ${package} but you are trying to create a snapshot for ${changelog_package}"
+  exit 1
+fi
+
+build_dir=$(basename $(pwd))
+
+sudo mock -r ${config} --bootstrap-chroot --isolation=simple --clean
+sudo mock -r ${config} --bootstrap-chroot --isolation=simple --init
+mock_root=$(sudo mock -r ${config} --bootstrap-chroot --isolation=simple --print-root-path)
+sudo mock -r ${config} --bootstrap-chroot --isolation=simple --install rpmdevtools git rpm-build cmake boost-devel openssl-devel cyrus-sasl-devel libbson-devel mongo-c-driver-devel snappy-devel gcc-c++ libzstd-devel
+sudo mock -r ${config} --bootstrap-chroot --isolation=simple --copyin "$(pwd)" "$(pwd)/${spec_file}" /tmp
+if [ ! -f build/VERSION_CURRENT ]; then
+  sudo mock -r ${config} --bootstrap-chroot --isolation=simple --cwd "/tmp/${build_dir}" --chroot -- /bin/sh -c "(
+    set -o xtrace ;
+    python3 etc/calc_release_version.py | sed -E 's/([^-]+).*/\1/' > build/VERSION_CURRENT
+    )"
+  sudo mock -r ${config} --bootstrap-chroot --isolation=simple --copyout "/tmp/${build_dir}/build/VERSION_CURRENT" build/
+fi
+
+bare_upstream_version=$(sed -E 's/([^-]+).*/\1/' build/VERSION_CURRENT)
+# Upstream version in the .spec file cannot have hyphen (-); replace the current
+# version so that the dist tarball version does not have a pre-release component
+sudo sh -c "echo ${bare_upstream_version} > build/VERSION_CURRENT"
+echo "Found bare upstream version: ${bare_upstream_version}"
+git_rev="$(git rev-parse --short HEAD)"
+snapshot_version="${bare_upstream_version}-0.$(date +%Y%m%d)+git${git_rev}"
+echo "Upstream snapshot version: ${snapshot_version}"
+current_package_version=$(rpmspec --srpm -q --qf "%{version}-%{release}" ${spec_file})
+
+if [ -n "${current_package_version##*${git_rev}*}" ]; then
+  echo "Making RPM changelog entry"
+  sudo mock -r ${config} --bootstrap-chroot --isolation=simple --cwd "/tmp/${build_dir}" --chroot -- rpmdev-bumpspec --comment="Built from Git Snapshot." --userstring="Test User <test@example.com>" --new="${snapshot_version}%{?dist}" ${spec_file}
+fi
+
+sudo mock -r ${config} --bootstrap-chroot --isolation=simple --copyout "/tmp/${build_dir}/${spec_file}" ..
+
+sudo mock -r ${config} --bootstrap-chroot --isolation=simple --cwd "/tmp/${build_dir}" --chroot -- /bin/sh -c "(
+  set -o xtrace ;
+  [ -d build ] || mkdir build ;
+  cd build ;
+  /usr/bin/cmake -DCMAKE_BUILD_TYPE=Release -DBSONCXX_POLY_USE_BOOST=1 -DENABLE_UNINSTALL=OFF .. ;
```

Comment:
No need to change. Do you know off-hand if the Boost polyfill is necessary? I imagine latest Fedora has a new enough C++ compiler to support `-DCMAKE_CXX_STANDARD=17`.

Reply:
That's what I use for the Debian package build and I carried it over here as well.  It is probably OK to change, but I'd want to discuss the implications first.